src/rviz/message_filter_display.h

```diff
@@ -137,7 +142,17 @@ class MessageFilterDisplay: public _RosTopicDisplay
 
       try
       {
-        sub_.subscribe( update_nh_, topic_property_->getTopicStd(), 10 );
+        ros::TransportHints transport_hint;
+        // Determine UDP vs TCP transport for user selection.
+        if (unreliable_property_->getBool())
+        {
+          transport_hint = ros::TransportHints().unreliable();
+        }
+        else
```

Comment:
Else clause is superfluous, I think. Also, do you have to instantiate another one and copy? What about just:

```
       if (unreliable_property_->getBool())
       {
         transport_hint.unreliable();
       }
```